File: app/Models/Result/LiveboardSearchResult.php
<?php

namespace Irail\Models\Result;

use Irail\Models\Cachable;
use Irail\Models\DepartureOrArrival;
use Irail\Models\Station;

class LiveboardSearchResult implements Cachable
{
    use CachableResult;

    private Station $station;
    /**
     * @var $stops DepartureOrArrival[]
     */
    private array $stops;

    /**
     * @param Station $station
     * @param DepartureOrArrival[] $stops
     */
    public function __construct(Station $station, array $stops)
    {
        $this->station = $station;
        $this->stops = $stops;
    }

    /**
     * @return Station
     */
    public function getStation(): Station
    {
        return $this->station;
    }

    /**
     * @return DepartureOrArrival[]
     */
    public function getStops(): array
    {
        return $this->stops;
    }
}

File: src/main/java/be/irail/api/dto/result/LiveboardSearchResult.java
package be.irail.api.dto.result;

import be.irail.api.dto.DepartureOrArrival;
import be.irail.api.dto.StationDto;
import java.util.List;

/**
 * Result of a liveboard search for a specific station.
 * Contains the station information and a list of upcoming departures or arrivals.
 */
public class LiveboardSearchResult {
    private final StationDto station;
    private final List<DepartureOrArrival> stops;

    /**
     * Constructs a new LiveboardSearchResult.
     *
     * @param station the station for which the liveboard was requested
     * @param stops the list of vehicle stops found
     */
    public LiveboardSearchResult(StationDto station, List<DepartureOrArrival> stops) {
        this.station = station;
        this.stops = stops;
    }

    /**
     * Gets the station associated with this liveboard.
     * @return the station
     */
    public StationDto getStation() {
        return station;
    }

    /**
     * Gets the list of vehicle stops (departures or arrivals) at the station.
     * @return the list of stops
     */
    public List<DepartureOrArrival> getStops() {
        return stops;
    }
}
